packages/endo/README.md

```diff
@@ -0,0 +1,283 @@
+
+
+## Usage
+
+Endo runs your application in a single frozen realm, endowing only the main
+compartment with the given powerful global properties and built-in modules.
+
+The `importPath` function runs a compartmentalized application off the file
+system.
+The `endowments` are properties to add to the `globalThis` in the global scope
+of the application's main package compartment.
+The `modules` are built-in modules to grant the application's main package
+compartment.
+
+```js
+import "ses";
+import fs from "fs";
+import { importPath } from "endo";
+
+lockdown();
+
+// ...
+
+const modules = { fs };
+const endowments = { console };
+
+const { namespace } = await importPath(
+  fs.promises.readFile,
+  modulePath,
+  endowments,
+  modules
+);
+```
+
+> TODO
+>
+> A future version will allow application authors to distribute
+> their global endowments and granted built-in modules to third-party
+> packages within the application, à la [LavaMoat].
+
+The `importPath` function uses `loadPath`.
+Using `loadPath` directly allows for deferred execution or multiple runs with
+different endowments or modules.
+
+Use `writeArchive` to capture an application in an archival format.
+
+```js
+import fs from "fs";
+import { writeArchive } from "endo";
+
+await writeArchive(
+  fs.promises.writeFile,
+  fs.promises.readFile,
+  'app.agar', // the archive to write
+  'app.js' // the application to capture
+);
+```
+
+The `writeArchive` function uses `makeArchive`.
+Using `makeArchive` directly gives you the archive bytes.
+
+Use `importArchive` to run an application from an archive.
+Note the similarity to `importPath`.
+
+```js
+import "ses";
+import fs from "fs";
+import { importPath } from "endo";
+
+lockdown();
+
+// ...
+
+const modules = { fs };
+const endowments = { console };
+
+const { namespace } = await importArchive(
+  fs.promises.readFile,
+  archivePath,
+  endowments,
+  modules
+);
+```
+
+The `importArchive` function composes `loadArchive` and `parseArchive`.
+Use `loadArchive` to defer execution or run multiple times with varying
+endowments.
+Use `parseArchive` if to construct a runner from the bytes of an archive.
+
+
+## Ruminations on the Name
+
+* In Latin, "endo-" means "internal" or "within".
+  This is fitting because Endo runs Node _within_ a safe sandbox.
+  This is fitting in turn because Endo is built on the legacy of Google Caja.
+  In Spanish, "caja" means "box" and is related to the Latin word "capsum" and
+  English "capsule", as in "encapsulate".
+* Endo is an anagram of Node and Deno.
+  That is to say, we are not Done yet.
+* The `endo` command, like the `sudo` command, is a "do" command.
+  However, instead of escalating priviliedge, it encapsulates priviledge.
+* So, "E.N.Do" forms the acronym "Encapsulated Node Do".
+
+So, just as "soo-doo" (super user do) and "soo-doh" (like "pseudo") are valid
+pronunciations of `sudo`, "en-doh" and "en-doo" are both valid pronunciations of
```

Comment:
argh, there went my headcanon (en-doh). I thought "at last, a program with an unambiguous pronounciation". Bah :)

Reply:
Totally have to mention "endo" vs. "endow".

Reply:
This is why we have reviews!